spec/services/monthly_district_data_service_spec.rb

```diff
@@ -0,0 +1,72 @@
+require "rails_helper"
+
+RSpec.describe MonthlyDistrictDataService do
+  let(:organization) { FactoryBot.create(:organization) }
+  let(:facility_group) { create(:facility_group, organization: organization) }
+  let(:facility) { create(:facility, facility_group: facility_group) }
+  let(:region) { facility.region.district_region }
+  let(:hypertensive_patient) { create(:patient, :hypertension, assigned_facility: facility) }
+  let(:period) { Period.month(Date.today) }
+  let(:service) { described_class.new(region, period) }
+
+  def find_in_csv(csv_data, row_index, column_name)
+    headers = csv_data[2]
+    column = headers.index(column_name)
+    csv_data[row_index][column]
+  end
+
+  describe "#report" do
+    it "produces valid csv data" do
+      result = service.report
+      expect {
+        CSV.parse(result)
+      }.not_to raise_error
+    end
+
+    it "provides accurate numbers for the district" do
```

Comment:
These are poor specs. I got hit by another last minute issue and the data setup for these tests would have taken me a while to figure out.

Reply:
Related to the [previous comment](https://github.com/simpledotorg/simple-server/pull/2458/files#r625529751), I would suggest having an intermediate representation of the report in a hash that we can test easily for correctness. The CSV test could then just ensure that the value/header mapping is correct.